src/main/java/karuna/karuna_backend/Authentication/AuthController.java

```diff
@@ -0,0 +1,104 @@
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@RestController
+@AllArgsConstructor
+public class AuthController {
+
+    private final UserService service;
+    private final JwtTokenService jwtTokenService;
+    private final JwtConfig jwtConfig;
+
+    @PostMapping("/login")
+    @Operation(summary = "Validates the provided authentication token",
+            description = "Checks if the token is valid and not expired. Returns token status.")
+    @ApiResponse(responseCode = "200", description = "Token is valid",
+            content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = UserDTO.class)))
+    @ApiResponse(responseCode = "401", description = "Credentials provided by user are invalid",
+            content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = AuthenticationErrorResponse.class)))
+    public ResponseEntity<?> authenticateUser(@RequestBody LoginRequestDto loginRequestDto) {
+        //TODO: Sanitize inputs, XSS prevention
+        String jwt = service.authenticateUser(loginRequestDto.username(), loginRequestDto.password());
+        LoginResponseDto body = LoginResponseDto.builder()
+                .username(jwtTokenService.getSubject(jwt))
+                .tokenExpirationTime(jwtTokenService.getExpirationDate(jwt))
+                .build();
+
+        return createJwtResponse(jwt, body, HttpStatus.OK);
+    }
+
+
+    @PostMapping("/register")
+    @Operation(summary = "Validates the provided authentication token",
+            description = "Checks if the token is valid and not expired. Returns token status.")
+    @ApiResponse(responseCode = "200", description = "Token is valid",
+            content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = UserDTO.class)))
+    @ApiResponse(responseCode = "409", description = "One of constraints was violated",
+            content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = DataIntegrityErrorResponse.class)))
+    public ResponseEntity<?> registerUser(@RequestBody RegisterRequestDto registerRequestDto) {
+
+        String jwt = service.registerUser(registerRequestDto.username(), registerRequestDto.password());
+        RegisterResponseDto body = RegisterResponseDto.builder()
+                .username(jwtTokenService.getSubject(jwt))
+                .tokenExpirationTime(jwtTokenService.getExpirationDate(jwt))
+                .build();
+
+        return createJwtResponse(jwt, body, HttpStatus.CREATED);
+    }
+
+
+
+    //TODO: This endpoint will be completely removed
```

Comment:
Mayby we remove it in this pull request?

Reply:
Not really becasue that was whole point of this issue, to let frontend communicate with backend and this testEndpoint